Sources/SwiftKueryPostgreSQL/PostgreSQLConnection.swift

```diff
@@ -286,16 +295,21 @@ public class PostgreSQLConnection: Connection {
     }
     
     private func prepareStatement(name: String, for query: String) -> String? {
+        if let error = setUpForRunningQuery() {
+            return error
+        }
         let result = PQprepare(connection, name, query, 0, nil)
         let status = PQresultStatus(result)
         if status != PGRES_COMMAND_OK {
-                var errorMessage = "Failed to create prepared statement."
-                if let error = String(validatingUTF8: PQerrorMessage(connection)) {
-                    errorMessage += " Error: \(error)."
-                }
-                PQclear(result)
-                return errorMessage
+            setState(.idle)
```

Comment:
Since `setState(.idle)` is the first thing done regardless of which path we take, it would be clearer if there was a single call, moved before the `if status`.